### PR TITLE
Search in navbar works only for first search term

### PR DESCRIPTION
**Pull request by kravets-levko**
## What type of PR is this? (check all applicable)

- [x] Bug Fix

## Related Tickets & Documents

Fixes getredash/redash#4856

### CODE REVIEW

File: client/app/pages/queries-list/QueriesList.jsx [hunk 1]
```diff
@@ -86,6 +87,22 @@ class QueriesList extends React.Component {
     }),
   ];
 
+  componentDidMount() {
```
Comment (arikfr) [inline client/app/pages/queries-list/QueriesList.jsx:90]: Shouldn't the component be re-rendered when the URL updates?

Comment (kravets-levko) [reply client/app/pages/queries-list/QueriesList.jsx:90]: Well, no. Router will recreate controller only if path changes. Watching for url parameters is responsibility of each controller

Comment (arikfr) [reply client/app/pages/queries-list/QueriesList.jsx:90]: This is not how I would expect a router to work in React and in general it leads to repeating this logic anywhere where we might be using query string parameters. 

I would've expected the page to receive the parameters as props, which would trigger React's rendering when/if needed.

Was there some explicit reason to make it work this way?

File: client/app/components/items-list/components/Sidebar.jsx [hunk 2]
```diff
@@ -11,15 +11,25 @@ import TagsList from "@/components/TagsList";
...
+  }, [value]);
+
+  const onInputChange = useCallback(
+    event => {
+      const newValue = event.target.value;
+      setCurrentValue(newValue);
+      onChange(newValue);
+    },
+    [onChange]
+  );
```
Comment (arikfr) [inline client/app/components/items-list/components/Sidebar.jsx:27]: Why is this needed? It seems that if we set `InputControl`'s `value` as the received `value` prop it will work. Does it create a loop in some cases or something like this?

File: client/app/pages/queries-list/QueriesList.jsx (hunk 1 above)
Comment (arikfr) [reply client/app/pages/queries-list/QueriesList.jsx:90]: (this is a general discussion, not something necessary to address in this PR)

File: client/app/components/items-list/components/Sidebar.jsx (hunk 2 above)
Comment (kravets-levko) [reply client/app/components/items-list/components/Sidebar.jsx:27]: `onChange` is debounced, so if passing `value` prop directly to `Input` component it will update with delay (user will type and see typed text only after they stop typing). This is quite common pattern - use local state to reflect typed text immediately, and sync it with prop when prop changes. Also, `defaultValue` doesn't work in this case because it will be used only when `Input` is mounted and all subsequent updates will be ignored.

File: client/app/pages/queries-list/QueriesList.jsx (hunk 1 above)
Comment (kravets-levko) [reply client/app/pages/queries-list/QueriesList.jsx:90]: > Was there some explicit reason to make it work this way?

That was a sort of optimization - to make sure that router will reload controllers only when needed. At that point, it seemed a good idea to bind controllers only to URL path so if path doesn't change - the same controller is kept. But now I agree with your point and I think we need to revisit this mechanism. It will require some changes to `ItemsList` component as I guess only controllers based on it use URL parameters. I'll create an issue for this under a `tech debt` label

Comment (kravets-levko) [reply client/app/pages/queries-list/QueriesList.jsx:90]: Issue getredash/redash#4864
